pkg/router/template/router.go

```diff
@@ -172,12 +173,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	if len(alias.PreferPort) == 0 {
+		for i := range svc.EndpointTable {
+			endpoint := svc.EndpointTable[i]
```

Comment:
Assuming this is necessary (it's weird to me you are walking this table twice, but I haven't actually looked at it) I think the idiomatic go way to update the values stored in items in a slice would be:
```
for i := range svc.EndpointTable {
    endpoint := &svc.EndpointTable[i]
    hash := $WHATEVER
    endpoint.IDHash = fmt.Sprintf(....hash)
}
```
Notice endpoint is a pointer to the value.

Now if I were going to write it your way, because I didn't really understand or want to use pointers but did want to make use of the return values of `range` it should look like:
```
for i, endpoint := range svc.EndpointTable {
    hash := $WHATEVER
    svc.EndpointTable[i].IDhash = fmt.Sprintf("%x", hash)
}
```